stac_fastapi/api/stac_fastapi/api/app.py

```diff
@@ -484,7 +479,7 @@ def __attrs_post_init__(self):
 
         # add middlewares
         for middleware in self.middlewares:
-            self.add_middleware(middleware)
+            self.app.user_middleware.insert(0, middleware)
```

Comment:
@vincentsarago why use the internal user_middleware list here, and not just use `add_middleware`? Do we need to add middleware after a __call__ is made to the FastAPI app, so you're trying to avoid an error with registering middleware after the initialization? That seems like it might not be a good pattern, to register middleware after initialization; if we aren't supporting that, thne `self.app.add_middleware(middleware)` seems better here.

Reply:
good call, let's use `self.app.add_middleware`

🙏 thanks @lossyrob

Reply:
ohhhh I see why we can't use starlette's add_middleware.

The `add_middleware` method expect a list of Callable + arguments (which will be turned into Middleware object) https://github.com/encode/starlette/blob/5a1bec33f8d6a669a3670f51034de83292d19408/starlette/applications.py#L134-L142

But the `StacApi.middlewares` is a list of *Middleware* objects so we need to insert them into the `user_middleware` list ourself. 

That's said, I think it would be good to raise the same error from starlette https://github.com/encode/starlette/blob/5a1bec33f8d6a669a3670f51034de83292d19408/starlette/applications.py#L140-L141